audeer/core/io.py

```diff
@@ -420,14 +428,15 @@ def list_file_names(
         list of path(s) to file(s)
 
     Raises:
-        NotADirectoryError: if ``os.path.dirname(path)``
-            is not a directory
-        FileNotFoundError: if ``os.path.dirname(path)``
-            does not exists
+        NotADirectoryError: if ``path`` is a directory
+            that does not exist,
+            or ``path`` is a pattern
+            and ``os.dirname(path)`` does not exist
 
     Examples:
         >>> dir_path = mkdir('path')
         >>> _ = touch(os.path.join(dir_path, 'file.wav'))
+        >>> _ = touch(os.path.join(dir_path, 'File.wav'))
         >>> _ = touch(os.path.join(dir_path, '.lock'))
         >>> sub_dir_path = mkdir(os.path.join('path', 'sub'))
         >>> _ = touch(os.path.join(sub_dir_path, 'file.ogg'))
```

Comment:
In all those examples above we use currently `os.path.join()` as `path()` from `audeer` is not available as it is imported as `safe_path` so we can use the variable `path` in the source code.

If we want to promote `path()` here in the docstring examples we could maybe add it to the docstring namespace similar to what we do with `pd` in https://github.com/audeering/audmath/blob/main/audmath/core/conftest.py, but I'm not completely sure if this will make it too complicated to understand for new developers afterwards.

Reply:
If we think this is important, I suggest we do it in a separate PR as probably also other examples are affected. Another solution would be to simplify the example and to create the files in the fixture instead.

Reply:
I created https://github.com/audeering/audeer/issues/95 to track this.